daemon/runtime_unix.go

```diff
@@ -108,6 +128,9 @@ func (daemon *Daemon) initRuntimes(runtimes map[string]types.Runtime) (err error
 		if rt.Shim == nil {
 			rt.Shim = defaultV2ShimConfig(daemon.configStore, rt.Path)
 		}
+		if rt.Shim.Opts == nil {
```

Comment:
Only the runc shims takes these default options, if the options are nil then we need to leave it nil unless we are using a runc shim.